File: src/Main/java/com/nishant/algorithms/Sorting/BubbleSort/BubbleSort.java
package com.nishant.algorithms.Sorting.BubbleSort;

public class BubbleSort {

    public static void sort(int[] arr) {
        boolean didFind = true;

        for(int end = arr.length; end > 0 && didFind; end --) {
            didFind = false;

            for(int i = 0; i < end - 1; i ++) {
                if(arr[i] > arr[i + 1]) {
                    swap(arr, i, i + 1);
                    didFind = true;
                }
            }
        }
    }

    private static void swap(int[] arr, int pos1, int pos2) {
        int temp = arr[pos1];
        arr[pos1] = arr[pos2];
        arr[pos2] = temp;
    }

}

File: src/main/python/com/nishant/sorting/bubblesort/BubbleSort.py
def bubble_sort(arr):
    did_find = True
    end = len(arr)

    while end >= 0:
        if did_find == False:
            break

        did_find = False

        for i in range(end - 1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                did_find = True

        end = end - 1

if __name__ == "__main__":
    arr = [1, 2, 6, 4, 5]
    print(str(arr))
    bubble_sort(arr)
    print(str(arr))
